modules/nf-core/falco/main.nf

```diff
@@ -0,0 +1,59 @@
+process FALCO {
+    tag "$meta.id"
+    label 'process_medium'
+
+
+    conda (params.enable_conda ? "bioconda::falco=1.2.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/falco:1.2.1--h867801b_3':
+        'quay.io/biocontainers/falco:1.2.1--h867801b_3' }"
+
+    input:
+    tuple val(meta), path(reads)
+
+    output:
+    tuple val(meta), path("*.html"), emit: html
+    tuple val(meta), path("*.txt") , emit: txt
+    path  "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    if (meta.single_end) {
```

Comment:
Here I would check the size of the `reads` input. If it's more than one file, it should go to the second case, regardless of it being PE or SE. Endedness has no effect on the tool

Reply:
true, but why you want to analyze 30 files at once? there is no multi-threading here

Reply:
To each it's own I guess 🤷🏻  I don't see why anyone would, but we can't force people to think like we do.
Either way, it's more correct, since the current situation would break support for interleaved files for example

Reply:
aha. So which change do you propose?

Reply:
Ok, should it be in single end so? Another thing, Falco with one file will produce a standard output name. So you need to enforce the output name using -R, -D and -S. If you use more files those options will trigger an error. So this is the reason I put them only in single end. Should we indicate another meta info? Something that is not single end or paired end?

Reply:
SE or PE don't matter for the tool. What you're doing with the settings is good. If there's one file, you need to set the naming options. If there's more than one file, you don't. Hence the check for the number of files instead of the endedness

Reply:
But since this is a replacement for fastqc this will break the compatibility. People should be able to replace fastqc in the existing pipelines without changing the meta etc. Or you want I just ignore the meta and count the files? Not so sure is a good idea actually. Can we discuss on slack with more people, please? :)

Reply:
I don't think this will break anything, since fastq's also doesn't take endedness into account (I think).
Lets take this to slack!